crates/common/src/signer/schemes/ecdsa.rs

```diff
@@ -0,0 +1,229 @@
+use core::fmt;
+use std::hash::Hash;
+
+use derive_more::derive::{Deref, From, Into};
+use generic_array::GenericArray;
+use k256::ecdsa::{Signature as EcdsaSignatureInner, VerifyingKey as EcdsaPublicKeyInner};
+use serde::{Deserialize, Serialize};
+use serde_utils::hex;
+use ssz_types::{
+    typenum::{U33, U64},
+    FixedVector,
+};
+use tree_hash::TreeHash;
+
+use crate::{signature::compute_signing_root, types::Chain};
+
+pub type EcdsaSecretKey = k256::ecdsa::SigningKey;
+
+type CompressedPublicKey = [u8; 33];
+
+#[derive(Debug, Clone, Copy, From, Into, Serialize, Deserialize, PartialEq, Eq, Deref, Hash)]
+#[serde(transparent)]
+pub struct EcdsaPublicKey {
+    #[serde(with = "alloy::hex::serde")]
+    encoded: CompressedPublicKey,
+}
+
+impl EcdsaPublicKey {
+    /// Size of the public key in bytes. We store the SEC1 encoded affine point
+    /// compressed, thus 33 bytes.
+    const SIZE: usize = 33;
+}
+
+impl Default for EcdsaPublicKey {
+    fn default() -> Self {
+        Self { encoded: [0; Self::SIZE] }
+    }
+}
+
+impl TreeHash for EcdsaPublicKey {
+    fn tree_hash_type() -> tree_hash::TreeHashType {
+        tree_hash::TreeHashType::Vector
+    }
+
+    fn tree_hash_packed_encoding(&self) -> tree_hash::PackedEncoding {
+        unreachable!("Vector should never be packed.")
+    }
+
+    fn tree_hash_packing_factor() -> usize {
+        unreachable!("Vector should never be packed.")
+    }
+
+    fn tree_hash_root(&self) -> tree_hash::Hash256 {
+        // NOTE:
+        // Unnecessary copying into a `FixedVector` just for its `tree_hash_root`
+        // implementation.  If this becomes a performance issue, we could use
+        // `ssz_types::tree_hash::vec_tree_hash_root`,  which is unfortunately
+        // not public.
+        let vec = self.encoded.to_vec();
+        FixedVector::<u8, U33>::from(vec).tree_hash_root()
+    }
+}
+
+impl ssz::Encode for EcdsaPublicKey {
```

Comment:
nit: leave as is but rust compiler in general is smart enough to inline the one liner function that return value